.github/workflows/ci.yml

```diff
@@ -0,0 +1,93 @@
+####################################################################################################
+# Continuous Integration [Lint, Test, Build] on All Pull-Requests + Push to Master
+####################################################################################################
+# References:
+# - https://github.com/snok/install-poetry
+# - https://github.com/actions/setup-python#caching-packages-dependencies
+# - https://stackoverflow.com/questions/62977821/how-to-cache-poetry-install-for-github-actions
+# - https://github.com/codecov/codecov-action
+####################################################################################################
+name: build
+
+on:
+  push:
+    branches: ["master"]
+  pull_request:
+
+jobs:
+  ci:
+    #----------------------------------------------
+    # -----  setup operating system (os)  -----
+    #----------------------------------------------
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        # TODO(amir): add  "ubuntu-latest", "windows-latest"
```

Comment:
currently, there is an issue with `Linux`:
- https://stackoverflow.com/questions/52244164/importing-pyautogui-in-ubuntu-throwing-keyerror-display
- https://command-not-found.com/xhost
we should first fix the `os` compatibility; then add the linux to ci.